src/lib.ts

```diff
@@ -229,7 +229,9 @@ export function canAcquireEffect(ef: Effect): boolean {
         case "synthesize":
           return false; // We currently don't support sweet synthesis
         case "barrelprayer":
-          return get("barrelShrineUnlocked") && !get("_barrelPrayer");
+          return (
+            get("barrelShrineUnlocked") && !get("_barrelPrayer") && myClass() === $class`Sauceror`
+          );
```

Comment:
This doesn't seem correct. We can also get the `barrel chested` (SC, +150%wdmg), `pork barrel` (PM, +90%item), `beer barrel polka` (AT, +45%booze) or even `barrel of laughs` (TT, +90hp) when calling `tryAcquiringEffect()` on said effects, and said effects are indeed supported by the script.

Reply:
Specifically, it's the excluded buffs not excluding buffs from other classes - as reported by LP back in December. The better fix is probably in explicitlyExcludedBuffs, huh?

Reply:
Which you already resolved.